# FastQC Guide
FastQC is an important step....



# Install

# Usage

## Params


# Results
Important to look at results from fastQC as it might suggest concerns about the data that will influence downstream processing and analysis of data. Here are some examples from RNASeq of naked mole-rat blood samples. Contact James Gilbert for more details about the samples - will add link if/when they are publicly available.

More detail available [here](https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon/lessons/qc_fastqc_assessment.html).

## Basic Stats
It is important to note the number of sequences per sample, and check the sequence length and GC content are as expected.

![Basic stats](https://github.com/jdgilbert245/test-RNASeq-repo/blob/main/assets/basic-stats.png).

## Per Base Quality
You can confirm that per base quality scores are acceptable. Above 28 is considered high. For reads generated by Illumina sequencing, quality scores sometimes fall further along the read. Our reads were high quality at all positions- good for the analysis but not great for troubleshooting!

![per base sequence quality](https://github.com/jdgilbert245/test-RNASeq-repo/blob/main/assets/per-base-seq-quality.png)

## Per Sequence Quality
This is a density plot of the average quality score for each sequence. We are looking for the general quality to be â‰¥ Q30, which corresponds to a base-calling accuracy of 99.9%. We also look for secondary peaks at lower values. Ours again were pretty good.

![per sequence quality](https://github.com/jdgilbert245/test-RNASeq-repo/blob/main/assets/per-seq-quality.png)









